.github/workflows/sync-api-vmlinux-cron.yaml

```diff
@@ -0,0 +1,117 @@
+          - arch: arm64
+            tumbleweed_repo: 'https://download.opensuse.org/ports/aarch64/tumbleweed/repo/oss/aarch64/'
+    steps:
+      - name: Install bpftool
+        uses: mtardy/setup-bpftool@9bc044304197616a6321d452e04cc059e006d8e2 # v1.0.4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Download most recent kernel package from opensuse tumbleweed
+        run: |
+          image=$(wget -qO- ${{ matrix.tumbleweed_repo }} | grep "kernel-default-base-*" | head -n1 | awk '{split($0,file,"\""); print file[4]}')
+          wget -O image.rpm ${{ matrix.tumbleweed_repo }}$image
+          mkdir image && cd image
+          rpm2cpio ../image.rpm | cpio -idmv
+          prefix="./kernel-default-base-"
+          suffix=".rpm"
+          version=${image#"$prefix"}
+          version=${version%"$suffix"}
+          echo "KERNEL_VERSION=$version" >> $GITHUB_ENV
+
+      - name: Download extract-kernel script from upstream linux repo
+        run: |
+          wget -O extract-vmlinux https://raw.githubusercontent.com/torvalds/linux/refs/heads/master/scripts/extract-vmlinux
+          chmod +x extract-vmlinux
+
+      - name: Extract vmlinuz
+        run: |
+          ./extract-vmlinux image/usr/lib/modules/*/vmlinux.xz > decomp-vmlinuz
+
+      - name: Generate updated vmlinux_generated
+        run: |
+          mkdir out
+          bpftool btf dump file decomp-vmlinuz format c > out/vmlinux_generated_${{ matrix.arch }}.h
+          # Add an autogenerated comment
+          sed -i '1i// Autogenerated from opensuse tumbleweed ${{ env.KERNEL_VERSION }} kernel' out/vmlinux_generated_${{ matrix.arch }}.h
+          sed -i 's/__VMLINUX_H__/__VMLINUX_GENERATED_H__/g' out/vmlinux_generated_${{ matrix.arch }}.h
+
+      - name: Sync bpf_helper_defs.h from upstream libbpf repo (x86 only)
+        if: matrix.arch == 'x86'
+        env:
+          PATCH_LONG_INT: 'map_update_elem;map_delete_elem;probe_read;perf_event_output;probe_read_str;probe_read_user;probe_read_kernel'
+        run: |
+          wget -O out/bpf_helper_defs.h https://raw.githubusercontent.com/libbpf/libbpf/refs/heads/master/src/bpf_helper_defs.h
+          # Keep only helpers declarations
+          sed -i -n -e '/^static/p' out/bpf_helper_defs.h
+          # drop `bpf_` prefix from all helpers, since we don't use it
+          sed -i 's/const bpf_/const /g' out/bpf_helper_defs.h
+          # add __maybe_unused
+          sed -i 's/) =/) __maybe_unused =/g' out/bpf_helper_defs.h
+          # Add an autogenerated comment
+          sed -i '1i// Autogenerated from libbpf sources' out/bpf_helper_defs.h
+          # Patch various long return value to old signatures (int)
+          # to avoid eBPF verifier issues on old kernels
+          IFS=';' read -r -a to_be_patched <<< "${{ env.PATCH_LONG_INT }}"
+          for element in "${to_be_patched[@]}"; do sed -i "s/static long (\* const $element)/static int (\* const $element)/g" out/bpf_helper_defs.h; done
```

Comment:
In my eyes, these steps seem brittle and easy to break in the future.

Given the above, I think it's worth taking a step back and asking, is this automation really needed?
What we have done so far is to update the headers as we use the relevant features in our code, which worked quite well, IMO. 

I'm CCing: @kevsecurity @olsajiri @mtardy for additional thoughts.

Reply:
I don't really understand the aim of automatically updating either the helpers or the vmlinux.h. I don't think it is too much trouble to update either as required, but if a tool would help, then great. I just don't really see this as a fully-automated task.

Reply:
I have no strong opinion on this one; i created the PR to fix https://github.com/cilium/tetragon/issues/3306.
If maintainers feel like this is not really a thing they want, i am :100: on board with that :+1: